cargo-espflash/src/main.rs

```diff
@@ -192,27 +160,14 @@ fn build(
         None => {}
     }
 
-    let mut command = match tool {
-        "cargo" | "xbuild" => Command::new("cargo"),
-        "xargo" => Command::new("xargo"),
-        _ => unreachable!(),
-    };
-
-    let command = match tool {
-        "xargo" | "cargo" => command.arg("build"),
-        "xbuild" => command.arg("xbuild"),
-        _ => unreachable!(),
-    };
-
-    if "cargo" == tool && !has_build_std(".") {
+    if !has_build_std(".") {
         println!("NOTE: --tool cargo currently requires the unstable build-std, ensure .cargo/config{{.toml}} has the appropriate options.");
-        println!("See: https://doc.rust-lang.org/cargo/reference/unstable.html#build-std")
+        println!("See: https://doc.rust-lang.org/cargo/reference/unstable.html#build-std");
+        // TODO early exit here
```

Comment:
If you add the following lines at the top of the file:

```rust
#[cfg(unix)]
use std::os::unix::process::ExitStatusExt;
#[cfg(not(unix)]
use std::os::windows::process::ExitStatusExit;
```

Then you can return using:

```rust
exit_with_process_status(ExitStatus::from_raw(1))
```

Alternatively, it may be enough just to call `std::process::exit(1)` instead of using these wrapper functions, I can't actually remember why I included them.

Reply:
Ahh I see you've resolved this in your other PR, please ignore :)